ner_telhub/widgets/graphing_widgets.py

```diff
@@ -183,22 +193,39 @@ def __init__(self, index, model: DataModelManager, format=Format.LINE):
         layout.addWidget(chart_view)
         self.setLayout(layout)
 
+    def getDataID(self, id):
+        if DATA_IDS.keys().__contains__(id):
+            dataID = id
+            return DATA_IDS[dataID]
+
     def edit_callback(self, state: GraphState):
         """
         Handles a changed state from the edit dialog box.
         """
         if self.state.format != state.format:
             pass # In the future, change graph type
         if self.state.data1 != state.data1:
-            self.remove_series("data1")
-            self.add_series("data1", state.data1)
+            datatype1 = str(self.getDataID(GraphState.datatype1))
+            self.remove_series(datatype1)   #Currently at state where legend displays data number
+            self.add_series(datatype1, state.data1)   #This sets the legend, need to change this to check master_mapping.py
```

Comment:
I have a couple suggestions for these lines:
1. We can get the data type string with a call to `self.model.getDataType(some id)` instead of the custom function
2. We should remove the old data type and add the new one, as right now we are removing the new and adding the new. The correct way would look like this:
```suggestion
           self.remove_series(self.model.getDataType(self.state.data1))
           self.add_series(self.model.getDataType(newState.data1), newState.data1)
```